saya/core/src/orchestrator/persistent.rs

```diff
@@ -108,8 +108,17 @@ where
             .build()
             .await
             .unwrap();
+
+        // Since the `Felt` type is wrapping (`Felt::MAX + 1 = 0`), there is not
+        // need for a special case for the genesis block, and `+1` works as expected.
+        //
+        // TODO: should we change to `settlement.next_block_number()` instead to always return `u64`?
```

Comment:
@xJonathanLEI I would tend to add the new method, or always manipulating the block number as a `Felt`, wdyt?